markdown.dtx

```diff
@@ -8943,6 +8943,204 @@ defaultOptions.texComments = false
 %</lua,lua-cli>
 %<*manual-options>
 
+#### Option `texMathDollars`
+
+`texMathDollars` (default value: `false`)
+
+% \fi
+% \begin{markdown}
+%
+% \Optitem[false]{texMathDollars}{\opt{true}, \opt{false}}
+%
+:    true
+
+     :  Enable the Pandoc `tex_math_dollars` syntax extension.
+
+        ```
+        inline math: $E=mc^2$
+
+        display math: $$E=mc^2$$
+        ```
+
+:    false
+
+     :  Disable the Pandoc `tex_math_dollars` syntax extension.
+
+% \end{markdown}
+% \iffalse
+
+##### Lua Module Example {.unnumbered}
+
+Using a text editor, create a text document named `document.tex` with the
+following content:
+``` tex
+\input markdown
+\input lmfonts
+\directlua{
+  local markdown = require("markdown")
+  local convert = markdown.new({texMathDollars = true})
+  local input =
+    "$E=mc^2$" .. newline .. newline ..
+    "$$\hat{f} \left ( \xi  \right )= \int_{-\infty}^{\infty} f\left ( x  \right ) e^{-i2\pi \xi x} dx$$"
+  tex.sprint(convert(input)) }
+\bye
+```````
+Then, invoke LuaTeX from the terminal:
+``` sh
+luatex document.tex
+```````
+A PDF document named `document.pdf` should be produced and contain the
+following text:
+
+> $E=mc^2$
+>
+> $$\hat{f} \left ( \xi  \right )= \int_{-\infty}^{\infty} f\left ( x  \right ) e^{-i2\pi \xi x} dx$$
+
+##### Lua CLI Example {.unnumbered}
+
+Using a text editor, create a text document named `document.tex` with the
+following content:
+``` tex
+\input markdown
+\input lmfonts
+\input optionfalse
+\par
+\input optiontrue
+\bye
+```````
+Using a text editor, create a text document named `content.md` with the
+following content:
+``` md
+$E=mc^2$
+
+$$\hat{f} \left ( \xi  \right )= \int_{-\infty}^{\infty} f\left ( x  \right ) e^{-i2\pi \xi x} dx$$
+``````
+Next, invoke LuaTeX from the terminal:
+``` sh
+texlua ⟨CLI pathname⟩ -- content.md optionfalse.tex
+texlua ⟨CLI pathname⟩ texMathDollars=true -- content.md optiontrue.tex
+luatex document.tex
+```````
+where \meta{CLI pathname} corresponds to the location of the Lua CLI script file,
+such as `~/texmf/scripts/markdown/markdown-cli.lua` on UN\*X systems or
+`C:\Users\`\meta{Your username}`\texmf\scripts\markdown\markdown-cli.lua` on Windows
+systems. Use the command `kpsewhich -a markdown-cli.lua` to locate the Lua CLI
+script file using [Kpathsea][].
+
+A PDF document named `document.pdf` should be produced and contain the
+following text:
+
+> \$E=mc^2\$
+> 
+> \$\$\hat{f} \left ( \xi  \right )= \int_{-\infty}^{\infty} f\left ( x  \right ) e^{-i2\pi \xi x} dx\$\$
+>
+> $E=mc^2$
+> 
+> $$\hat{f} \left ( \xi  \right )= \int_{-\infty}^{\infty} f\left ( x  \right ) e^{-i2\pi \xi x} dx$$
```

Comment:
Since the HTML user manual is produced via Pandoc, it is important to ensure that the output is correctly rendered by Pandoc. Here is the result of the conversion:
```
$ cat > example.md
> \$E=mc^2\$
> 
> $E=mc^2$
> $$\hat{f} \left ( \xi  \right )= \int_{-\infty}^{\infty} f\left ( x  \right ) e^{-i2\pi \xi x} dx$$
^D
$ pandoc -f markdown -t html example.md
<blockquote>
<p>$E=mc^2$</p>
<p>$$ ( )= _{-}^{} f( x ) e^{-i2x} dx$$</p>
<p><span class="math inline"><em>E</em> = <em>m</em><em>c</em><sup>2</sup></span></p>
<p><br /><span class="math display"><em>f̂</em>(<em>ξ</em>) = ∫<sub> − ∞</sub><sup>∞</sup><em>f</em>(<em>x</em>)<em>e</em><sup> − <em>i</em>2<em>π</em><em>ξ</em><em>x</em></sup><em>d</em><em>x</em></span><br /></p>
</blockquote>
```

It seems that Pandoc heuristically recognizes TeX commands as raw TeX input due to [the `raw_tex` extension][1] and does not include them in HTML output. Here is the output with `raw_tex` disabled:

```
$ pandoc -f markdown-raw_tex -t html example.md
<blockquote>
<p>$E=mc^2$</p>
<p>$$\hat{f} \left ( \xi \right )= \int_{-\infty}^{\infty} f\left ( x \right ) e^{-i2\pi \xi x} dx$$</p>
<p><span class="math inline"><em>E</em> = <em>m</em><em>c</em><sup>2</sup></span></p>
<p><br /><span class="math display"><em>f̂</em>(<em>ξ</em>) = ∫<sub> − ∞</sub><sup>∞</sup><em>f</em>(<em>x</em>)<em>e</em><sup> − <em>i</em>2<em>π</em><em>ξ</em><em>x</em></sup><em>d</em><em>x</em></span><br /></p>
</blockquote>
```

We should disable `raw_tex` by changing `pandoc -f markdown` to `pandoc -f markdown-raw_tex` in `Makefile` as well. 

 [1]: https://pandoc.org/MANUAL.html#extension-raw_tex